azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -992,60 +1015,52 @@ def _send_new_connect_op_down(self):
         def on_connect_complete(op, error):
             this = self_weakref()
             if this:
+                logger.debug(
+                    "{}({}): on_connect_complete error={} state={} never_conencted={} connected={} ".format(
+                        this.name,
+                        op.name,
+                        error,
+                        this.state,
+                        this.never_connected,
+                        this.pipeline_root.connected,
+                    )
+                )
                 if error:
-                    if this.state == ReconnectState.NEVER_CONNECTED:
-                        logger.info(
-                            "{}({}): error on first connection.  Not triggering reconnection".format(
-                                this.name, op.name
-                            )
-                        )
+                    if this.never_connected:
```

Comment:
>if this.never_connected: [](start = 20, length = 24)

this can also be put in the else bock right ? this does the same as the else block ... a comment can be added that the else block is executed on if `never_connected =True ` as well as for other non-transient errors